### DISCUSSION TITLE
Error Map/Reduce Index (Computed value)

### DISCUSSION THREAD

**Discussion by MohMehrnia** [ravendb/ravendb#18466] [General]
```
I Created Map/Reduce Index By below Code

// Map
from v in docs.VideoPlayUserInfos
select new {
    v.CourseID, 
    v.FileID, 
    v.UserID, 
    v.FullName, 
    v.CurrentTime, 
    ContentDuration = v.ContentDuration.TotalSeconds,
    Diff=0
}


// Reduce 

from r in results
group r by new { r.UserID, r.FullName,r.CourseID, r.ContentDuration, r.FileID } into g
select new {
   g.Key.UserID,
   g.Key.FullName,
   g.Key.CourseID,
   g.Key.ContentDuration,
   g.Key.FileID,
   CurrentTime = g.Max(x=>x.CurrentTime), 
    Diff = g.Key.ContentDuration - g.Max(x=>x.CurrentTime)
}
```

Index Created Very longtime running and end show me error and index not created

**Comment by grisha-kotler** [comment]
What is the error that you see? (full stacktrace)

**Comment by MohMehrnia** [comment]
process after very long time running (3 hours)
server is done and can not see exception

**Reply by ayende** [reply]
How many entries do you have in the index? `v.ContentDuration.TotalSeconds` looks like a `TimeSpan.TotalSeconds`. That is a `double` and may have a _lot_ of unique values. That may mean that we have to do a lot of book-keeping internally, since you are grouping by that value.

What is the total size of the `VideoPlayUserInfos` collection? 

Three hours to run may be possible, especially if you are running on a cloud disk.

**Reply by MohMehrnia** [reply]
300k documents

**Reply by ayende** [reply]
How many entries do you have in the index?

**Reply by MohMehrnia** [reply]
fixed, thanks

**Reply by ayende** [reply]
What was the fix?

**Reply by MohMehrnia** [reply]
i changed and index worked
`Diff = g.Key.ContentDuration == g.Max(x=>x.CurrentTime)`
